### PR TITLE
fix(): fixed slicegw health status for single cluster (#207)

### PR DESCRIPTION
**Pull request by gourishkb**
* fix(): fixed slicegw health status for single cluster

* fix(): fixed broken UTs

* fix(): fixed ITs

* fix(): handling multiple slicegw deployments

Reference PR https://github.com/kubeslice/worker-operator/pull/207 
allure report https://kubeslice.github.io/e2e-allure-reports/Kind-worker-operator-2023-04-01T16:48:11-release-guwahati-394/index.html 

---------

### CODE REVIEW

**Comment by gourishkb** [comment]
```Summarizing 2 Failures:
  [FAIL] Hub ClusterController Cluster health check [It] Should update cluster CR with compoent status as error for all
  /workspace/tests/hub/cluster_controller_test.go:327
  [FAIL] Hub ClusterController Cluster health check [It] Should update cluster CR with component status as normal when pods running``` these cases are related to cluster health events present in master.

File: pkg/hub/controllers/cluster/reconciler.go [hunk 1]
```diff
@@ -168,7 +166,7 @@ func (r *Reconciler) updateClusterInfo(ctx context.Context, cr *hubv1alpha1.Clus
 
 	// Populate NodeIPs if not already updated
 	// Only needed to do the initial update. Later updates will be done by node reconciler
-	if cr.Spec.NodeIPs == nil || len(cr.Spec.NodeIPs) == 0 {
+	if isValidNodeIpList(cr.Spec.NodeIPs) || cr.Spec.NodeIPs == nil || len(cr.Spec.NodeIPs) == 0 {
```
Comment (mridulgain) [inline pkg/hub/controllers/cluster/reconciler.go:169]: Wouldn't it panic if `cr.Spec.NodeIPs` is nil ?

File: pkg/hub/controllers/cluster/reconciler.go [hunk 2]
```diff
@@ -272,6 +270,15 @@ func (r *Reconciler) getCluster(ctx context.Context, req reconcile.Request) (*hu
 	return hubCluster, nil
 }
 
+func isValidNodeIpList(nodeIPs []string) bool {
+	for _, nodeIP := range nodeIPs {
+		if nodeIP == "" {
```
Comment (mridulgain) [inline pkg/hub/controllers/cluster/reconciler.go:275]: Do we consider `nodeIP == ""` to be valid, since we are returning true? ?

**Comment by kmjayadeep** [comment]
@gourishkb cluster health check tests were passing in https://github.com/kubeslice/worker-operator/pull/204. The failures maybe be due to recent changes.
Also  @richiesebastian Is there any change in github actions file? I don't see ITs here in this PR

**Comment by gourishkb** [comment]
closing this PR for https://github.com/kubeslice/worker-operator/pull/213

**Pull request closed without merging**